ldap/servers/slapd/back-ldbm/vlv_srch.c

```diff
@@ -792,32 +792,28 @@ vlvIndex_isVlvIndexEntry(Slapi_Entry *e)
 }
 
 /*
- * Create the filename for the index.
- * Extract all the alphanumeric characters from the descriptive name.
- * Convert to all lower case.
+ * Generate the vlv db file name from the vlv name.
+ * Return NULL if vlv name cannot be converted to db name
  */
-static int
-vlvIndex_createfilename(struct vlvIndex *pIndex, char **ppc)
-{
-    int filenameValid = 1;
-    unsigned int i;
-    char *p, *filename;
-    filename = slapi_ch_malloc(strlen(pIndex->vlv_name) + 1);
-    p = filename;
-    for (i = 0; i < strlen(pIndex->vlv_name); i++) {
-        if (isalnum(pIndex->vlv_name[i])) {
-            *p = TOLOWER(pIndex->vlv_name[i]);
-            p++;
+char *
+vlvIndex_build_filename(const char *vlvname)
+{
+    size_t len = strlen(vlvname);
+    size_t len_prefix = strlen(file_prefix);
+    char *filename = slapi_ch_malloc(len_prefix + len + 1);
+    char *pt = filename;
```

Comment:
you may return NULL immediately if vlvname is NULL

Reply:
vlvname cannot be NULL because vlvIndex_createfilename was using strlen(p->vlv_name) without checking for NULL